diff_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestCompare(t *testing.T) {
+	doc := []byte(`{
+    "key" : "value"
+  }`)
+
+	if compare(doc, doc) == false {
+		t.Errorf("compare assertion failed")
+	}
+
+}
+
+func TestDiff(t *testing.T) {
+	doc1 := []byte(`{
+  "key" : "value"
+  }`)
+
+	doc2 := []byte(`{
+    "key" : "value1"
+    }`)
+
+	patch, err := diff(doc1, doc2)
+
+	if err != nil {
+		t.Errorf("Error generating patch", err)
+	}
+
+	str := string(patch)
+	if str != `{"key":"value1"}` {
+		t.Errorf("diff assertion failed")
+	}
+}
```

Comment:
@vladoatanasov, looks like you have serious formatting issues here. But everything else looks good to me.